src/util/childProcess.js

```diff
@@ -1,27 +1,53 @@
-"use strict";
-
 const _ = require("lodash");
 const clc = require("cli-color");
 const EventEmitter = require("events").EventEmitter;
+const StreamSlicer = require('stream-slic3r');
 
 const logger = require("../logger");
 const logStamp = require("./logstamp");
 
 const MESSAGE = "message";
-const DATA = "data";
 const CLOSE = "close";
 
 const NOT_GOOD_ENUFF_ERROR_MESSAGE = "Connection refused! Is selenium server started?";
 const ADDED_ERROR_MESSAGE_CONTEXT = "If running on saucelabs, perhaps " +
     "you're out of capacity and should TRY RUN AGAIN LATER :)";
 
+// MAKE NIGHWATCH ERROR & WARN LOGS VISIBLE IN MAGELLAN STDOUT:
+// ------------------------------------------------------------------------------------------------
+// Currently the "ERROR" and "WARN" logs from nightwatch are suppressed when the "DEBUG" flag is
+// turned OFF.. and our customers ALWAYS use the "DEBUG" off, because if you turn it on your log
+// will be filled with base64 screenshot gobbledegook... Also, with the approach taken, we will
+// sell all the "ERRORED" selenium request/response logs in our magellan log
+//
+// relevant code for this feature:
+//   STDOUT_WHITE_LIST, SLICE_ON_TEXT, infoSlicer, isTextWhiteListed
+// ------------------------------------------------------------------------------------------------
+
+// if the "this.handler.stdout" stream of the childprocess does not 
+// include atleast one of these tokens then it will not be included in the "this.stdout"
+const STDOUT_WHITE_LIST = ['ERROR', 'WARN', 'Test Suite', '✖']
+
+// we slice the VERBOSE nighwatch stdout stream on the purple INFO text that has black background
+const SLICE_ON_TEXT = '\033[1;35m\033[40mINFO\033[0m'
```

Comment:
do you know if this works for both nightwatch .9 and 1.x?

Reply:
good question.

looking at their logger in the master:
https://github.com/nightwatchjs/nightwatch/blob/master/lib/utils/logger.js

it seems it should work just fine cause they havent changed the logger INFO coloring at all

Reply:
https://github.com/nightwatchjs/nightwatch/blob/a65f49e5cc3f6ac3c23eac8c9cbb3da35001c93a/lib/utils/logger.js#L142